### DISCUSSION TITLE
Create should no longer assume RunLocal

### DISCUSSION THREAD

**Discussion by ajohnstone-ks** [MarimerLLC/csla#2160] [Ideas]
Using Csla 5 and DI, we have an issue were Creates are still being assumed as RunLocal.  This becomes an issue for us as we'd like to Inject stuff that only exists in the server-side build our of BO assembly.  

Please see [here](https://github.com/ajohnstone-ks/CreateRunLocal) for an example.

I think that unless Csla finds a Create method and its explicitly marked as RunLocal, it should otherwise function like any other DP call.  Otherwise we will have to add server-only assembly references to our client, or we won't be able to use inject.  Neither is really great.

**Comment by ajohnstone-ks** [comment]
It looks like this happens with Creates which don't need any criteria parameters, because the BO will always contain a DataPortal_Create with RunLocal attached.  Maybe its time to obsolete this method?

**Reply by rockfordlhotka** [reply]
It should be currently flagged with a warning via an analyzer, and the intent is to remove it (or at least truly mark with `Obsolete) in CSLA 6.

Though I suppose the analyzer won't actually warn you if _you_ have no create and allow it to fall back to the base class, that's an oversight.
